osu.Game/Overlays/Mods/ModSelectOverlay.cs

```diff
@@ -494,19 +496,28 @@ private void updateMods()
                 MultiplierLabel.FadeColour(Color4.White, 200);
         }
 
+        private ScheduledDelegate sampleOnDelegate;
+        private ScheduledDelegate sampleOffDelegate;
+
         private void modButtonPressed(Mod selectedMod)
         {
             if (selectedMod != null)
             {
-                if (State.Value == Visibility.Visible) sampleOn?.Play();
+                // Fixes buzzing when multiple mods are selected in the same frame.
+                sampleOnDelegate?.Cancel();
+                if (State.Value == Visibility.Visible)
+                    sampleOnDelegate = Scheduler.Add(() => sampleOn?.Play());
```

Comment:
I guess it's not a huge deal but you could presumably use `Scheduler.AddOnce()` for this, which would eliminate the delegate cancellation?